### PR TITLE
PT-727 Publish the version of the service in the status object

### PR DESCRIPTION
**Pull request by ydrisR**
Modify the common object "Status" to add the version of the service.
Also add unit tests to prevent regressions

### CODE REVIEW

File: clients/status/status.go [hunk 1]
```diff
@@ -3,41 +3,41 @@ package status
...
 // Type status holds the status return from an http request.
 type Status struct {
-	Code   int    `json:"code"`
-	Error  *int   `json:"error,omitempty"`
-	Reason string `json:"reason"`
+	Code    int    `json:"code"`
+	Error   *int   `json:"error,omitempty"`
+	Reason  string `json:"reason"`
+	Version string `json:"version"`
 }
```
Comment (CedricPerotto) [inline clients/status/status.go]: If we don't want to embed the Version in all status struct (error status for ex.). We could Extend the Status struct only for API status.

i.e. 
```
type StatusApi struct{
	Status 
	Version string `json:"version"`
}
```
with the factory function NewStatusApi 
```
func NewStatusApi(status Status) StatusApi {
	s := StatusApi{Status : status, Version: version.Long()}
	return s
}
```
The NewStatusApi  would be only called for api.GetStatus calls.

**Review by CedricPerotto — CHANGES_REQUESTED** [review]

File: clients/status/status.go (hunk 1 above)
Comment (ydrisR) [reply clients/status/status.go]: 👍

File: clients/version/version.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+**/
+
+// Singleton version instance
+var instance version
+var once sync.Once
+
+// Variables to be injected at build time
+var (
+	ReleaseNumber string //Release number. i.e. 1.2.3
+	ShortCommit   string
```
Comment (BigFab) [inline clients/version/version.go]:
```suggestion
	ShortCommit   string //Short commit id. i.e. 048a8d4
```

File: clients/version/version.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	FullCommit    string //Full commit id. i.e. e0c73b95646559e9a3696d41711e918398d557fb
+)
+
+func GetVersion() version {
+	once.Do(func() {
```
Comment (BigFab) [inline clients/version/version.go:61]:
```suggestion
        //Ensure this is called once so version instance is created as a singleton
	once.Do(func() {
```

File: clients/version/version.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+	Initialization wrapper
+**/
+
+// Singleton version instance
+var instance version
+var once sync.Once
+
+// Variables to be injected at build time
```
Comment (BigFab) [inline clients/version/version.go:51]:
```suggestion
// Variables to be injected at build time
// E.g. go build -ldflags "-X $GO_COMMON_PATH/clients/version.ReleaseNumber=$VERSION"
```

**Review by BigFab — COMMENTED** [review]
LGTM although I would like to see some more comments that I suggested inline

**Review by BigFab — APPROVED** [review]
LGTM now :)

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**